bool traj_active
string[] points